python/pyspark/worker.py

```diff
@@ -1914,6 +1975,16 @@ def read_udfs(pickleSer, infile, eval_type):
         ):
             # Arrow cast for type coercion is disabled by default
             ser = ArrowStreamArrowUDFSerializer(timezone, safecheck, _assign_cols_by_name, False)
+        elif eval_type == PythonEvalType.SQL_ARROW_BATCHED_UDF and not use_legacy_pandas_udf_conversion(runner_conf):
+            input_types = ([f.dataType for f in _parse_datatype_json_string(utf8_deserializer.loads(infile))])
+            ser = ArrowBatchUDFSerializer(
+                False,
+                input_types,
+                "row",
+                True,
+                None,
```

Comment:
As `ArrowBatchUDFSerializer` is specific to this feature, we don't need these, except for `input_types`?